src/scene/data.ts

```diff
@@ -64,14 +66,17 @@ export function loadFromJSON() {
   return new Promise<DataState>(resolve => {
     reader.onloadend = () => {
       if (reader.readyState === FileReader.DONE) {
+        const defaultAppState = getDefaultAppState();
         let elements = [];
+        let appState = defaultAppState;
         try {
           const data = JSON.parse(reader.result as string);
           elements = data.elements || [];
+          appState = data.appState || defaultAppState;
```

Comment:
Can you do `{...defaultAppState, ...data.appState}` so that if we add more things to appState down the line, it doesn't end up without a value.